src/datepicker/datepicker.js

```diff
@@ -523,6 +580,10 @@ function ($compile, $parse, $document, $position, dateFilter, dateParser, datepi
         }
       }
       ngModel.$parsers.unshift(parseDate);
+      ngModel.$parsers.unshift(minLimitParse);
+      ngModel.$formatters.unshift(minLimitFormat);
+      ngModel.$parsers.unshift(maxLimitParse);
+      ngModel.$formatters.unshift(maxLimitFormat);
```

Comment:
This is a lot of parsers and formatters to be adding to the arrays.  Ideally we should only be adding one parser and formatter which executes all of the modifications we desire.